python/delegates/view_item_delegate.py

```diff
@@ -1158,13 +1158,6 @@ def editorEvent(self, event, model, option, index):
                 else:
                     widget.unsetCursor()
 
-            # Emit a data changed signal for the index to paint according to the mouse move
-            # event (e.g. hover selection).
-            # TODO: leverage the 'roles' parameter to emit only the relevant data roles
-            # to consider, for a more efficient update. Currently there are some issues using
-            # this parameter due to mismatch method signatures between versions.
-            model.dataChanged.emit(index, index)
```

Comment:
I wonder if we shouldn't continue issuing dataChanged() here after setting the role data here for the index.model().setData(index, new_check_state, check_state_role) element ?

Reply:
This update is actually part of the PR (that has been merged) https://github.com/shotgunsoftware/tk-framework-qtwidgets/pull/133/files

In short, this data change signal is not necessary here and was triggering more paint events than needed, which negatively affects performance